LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -237,15 +259,37 @@ public async Task<bool> InitializeAsync()
                     if (twin.Properties.Desired.Contains(TwinProperty.Deduplication))
                     {
                         var val = twin.Properties.Desired[TwinProperty.Deduplication].Value as string;
-                        Enum.TryParse<DeduplicationMode>(val, out DeduplicationMode mode);
+                        Enum.TryParse<DeduplicationMode>(val, true, out DeduplicationMode mode);
                         this.Deduplication = mode;
                     }
 
                     if (twin.Properties.Desired.Contains(TwinProperty.ClassType))
                     {
                         if (string.Equals("c", (string)twin.Properties.Desired[TwinProperty.ClassType], StringComparison.InvariantCultureIgnoreCase))
                         {
-                            this.classType = LoRaDeviceClassType.C;
+                            this.ClassType = LoRaDeviceClassType.C;
+                        }
+                    }
+
+                    if (twin.Properties.Reported.Contains(TwinProperty.PreferredGatewayID))
+                    {
+                        this.preferredGatewayID = new ChangeTrackingProperty<string>(TwinProperty.PreferredGatewayID, twin.Properties.Reported[TwinProperty.PreferredGatewayID].Value as string);
+                    }
+
+                    if (twin.Properties.Reported.Contains(TwinProperty.Region))
+                    {
+                        var regionValue = twin.Properties.Reported[TwinProperty.Region].Value as string;
+                        if (Enum.TryParse<LoRaRegion>(regionValue, true, out var loRaRegion))
+                        {
+                            if (Enum.IsDefined(typeof(LoRaRegion), loRaRegion))
```

Comment:
did you check if this is required? I would assume that if Enum.TryParse returns true, it will have to be defined?

Reply:
Unfortunately enums are bad. In my experience the following will return true:`Enum.TryParse<LoRaRegion>("3123", true, out var region)`

Reply:
Didn't know this. That's pretty bad. The IsDefined is also taking an object which will cause a boxing operation... very nice. :(